### ISSUE TITLE
new versions of nyc break tap test suite

### ISSUE THREAD

**Issue by bcoe** [bcoe/nyc#113]
the tap suite runs several tests on require-hooks itself, these seem to be failing on newer versions of nyc:

```shell
test/require-hooks.js ................................. 4/6 6s
  compile-to-js require hook > failing test with unmatched stack-trace
  not ok error happened in the *.faux file
    found: ''
    pattern: 'file: test/fixtures/using-require-hook.faux'
    at:
      file: test/require-hooks.js
      line: 9
      column: 1000
      function: verifyOutput
    source: |
      test('compile-to-js require hook', function (t) {
    stack: |
      verifyOutput (test/require-hooks.js:9:1000)
      ChildProcess.exithandler (child_process.js:204:5)
      maybeClose (internal/child_process.js:764:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
  
  compile-to-js require hook > failing test with unmatched stack-trace
  not ok error happened in the *.faux file
    found: >
      /Users/benjamincoe/bcoe/node-tap/test/fixtures/using-require-hook.js
    
    
    
          1..0
    
    
    
      not ok 1 -
      /Users/benjamincoe/bcoe/node-tap/test/fixtures/using-require-hook.js
    
    
    
        ---
    
    
    
        results:
    
    
    
          ok: false
    
    
    
          count: 0
    
    
    
          pass: 0
    
    
    
          plan:
    
    
    
            start: 1
    
    
    
            end: 0
    
    
    
            skipAll: true
    
    
    
        arguments:
    
    
    
          - /Users/benjamincoe/bcoe/node-tap/test/fixtures/using-require-hook.js
    
    
    
        exitCode: 1
    
    
    
        timeout: 240000
    
    
    
        command: /Users/benjamincoe/.nvm/versions/io.js/v3.2.0/bin/iojs
    
    
    
        file: /Users/benjamincoe/bcoe/node-tap/test/fixtures/using-require-hook.js
    
    
    
        ...
    
    
    
    
    
    
    
      1..1
    pattern: 'file: test/fixtures/using-require-hook.faux'
    at:
      file: test/require-hooks.js
      line: 9
      column: 1000
      function: verifyOutput
    source: |
      test('compile-to-js require hook', function (t) {
    stack: |
      verifyOutput (test/require-hooks.js:9:1000)
      ChildProcess.exithandler (child_process.js:204:5)
      maybeClose (internal/child_process.js:764:16)
      Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

I bet this is being caused by the overall of require-hook that we've been iterating on.

It would be great to get this fixed so that @isaacs can pull the new improved version into tap.

Any thoughts: @jamestalmage, @novemberborn?

**Comment by jamestalmage** [comment]
https://github.com/isaacs/node-tap/pull/197

I think you forgot to switch `exclude` to glob patterns, or you were running Node `5.2.0` - both can cause that problem. (the test suite fails on `5.2.0` before that update)

**Comment by jamestalmage** [comment]
closed via isaacs/node-tap#197

**Comment by bcoe** [comment]
\o/ great job debugging this @jamestalmage
